app-arch/createrepo_c/createrepo_c-0.20.1-r1.ebuild

```diff
@@ -0,0 +1,79 @@
+# Copyright 2020-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="C implementation of createrepo"
+HOMEPAGE="https://github.com/rpm-software-management/createrepo_c"
+if [[ ${PV} = 9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/rpm-software-management/createrepo_c.git"
+else
+	SRC_URI="https://github.com/rpm-software-management/createrepo_c/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-2"
+SLOT="0"
+
+IUSE="doc legacy test"
+RESTRICT="!test? ( test )"
+
+DEPEND="
+	app-arch/bzip2:=
+	app-arch/drpm
+	app-arch/rpm
+	app-arch/xz-utils
+	app-arch/zchunk
+	dev-db/sqlite:3
+	dev-libs/glib:2
+	dev-libs/libxml2
+	dev-libs/openssl:=
+	net-misc/curl
+	sys-apps/file
+	sys-libs/libmodulemd
+	sys-libs/zlib:=
+"
+
+RDEPEND="${DEPEND}"
+
+BDEPEND="
+	${DEPEND}
+	doc? ( app-doc/doxygen )
+"
+
+PATCHES=(
+	"${FILESDIR}/${P}-Include-rpm-rpmstring.h-for-rasprintf.patch"
+	"${FILESDIR}/${P}-use-gio-instead-of-cp.patch"
+)
+
+src_prepare() {
+	cmake_src_prepare
+}
```

Comment:
I do not believe src_prepare override here should be necessary?  Doesn't cmake_src_prepare get called by default?

Reply:
Good catch; shrapnel from when I was experimenting with python bindings and tests.

FWIW I was able to get python building and all tests passing using the distutils eclass, however it seems to be a mutually exclusive build - either C or Python stuff gets installed at the end. I'm going to chat to upstream and find out if I'm missing something, because it would be nice to have some importable python modules for scripting.

That'll be a `-r2` though.

Reply:
ACK, yes, would be nice, we're just invoking the CLI tool from our end, which suits our needs perfectly.